vendor/arm/mali/mali_profiler.cpp

```diff
@@ -594,22 +625,38 @@ uint64_t MaliProfiler::get_counter_value(mali_userspace::MaliCounterBlockName bl
 			// If an MMU counter is selected, sum the values over MMU slices
 			for (int i = 0; i < num_l2_slices_; i++)
 			{
-				sum += get_counters(block, i)[find_counter_index_by_name(block, name)];
+				const uint32_t *counter = get_counters(block, i);
+				if (!counter)
+				{
+					return -1;
```

Comment:
This function is `uint64_t`, but we have `return -1` here. Could this cause confusions in the future?

Reply:
Possibly, error propagation seems to be an issue that is yet to be resolved. I think we should decide on a pattern and try and use this more consistently throughout the project?